packages/flutter/lib/src/rendering/object.dart

```diff
@@ -1230,6 +1230,47 @@ abstract class RenderObject extends AbstractNode with DiagnosticableTreeMixin im
     });
   }
 
+  /// Whether this has been disposed.
+  ///
+  /// If asserts are disabled, this property is always null.
+  bool? get debugDisposed {
+    bool? disposed;
+    assert(() {
+      disposed = _debugDisposed;
+      return true;
+    }());
+    return disposed;
+  }
+
+  bool _debugDisposed = false;
+
+  /// Release any resources held by this render object.
+  ///
+  /// If this render object has created any children directly, it should dispose
+  /// of those children in this method as well. However, it should not
+  /// dispose of children that were created by some other object, such as
+  /// a [RenderObjectElement].
```

Comment:
Not sure I understand what is meant by this "However"?

Reply:
Tried to clarify. Basically, only dispose ROs you created directly, which might include your children but does not automatically include them.